File: src/services/diff.ts
import { invoke } from '@tauri-apps/api/core';

export async function unifiedPatch(
  left: string,
  right: string,
  context: number,
  nameA = 'original.txt',
  nameB = 'modified.txt',
): Promise<string> {
  return invoke<string>('diff_unified_patch', {
    left,
    right,
    context,
    nameA,
    nameB,
  });
}

File: src-tauri/src/commands/diff.rs
use similar::TextDiff;

#[tauri::command]
pub fn diff_unified_patch(
    left: String,
    right: String,
    context: u32,
    name_a: Option<String>,
    name_b: Option<String>,
) -> Result<String, String> {
    let diff = TextDiff::from_lines(&left, &right);
    let name_a = name_a.unwrap_or_else(|| "original.txt".into());
    let name_b = name_b.unwrap_or_else(|| "modified.txt".into());
    let patch = diff
        .unified_diff()
        .context_radius(context as usize)
        .header(&name_a, &name_b)
        .to_string();
    Ok(patch)
}
